src/internal.c

```diff
@@ -14851,7 +14851,9 @@ static word32 QSH_KeyExchangeWrite(WOLFSSL* ssl, byte isServer)
                     byte* es  = encSecret;
                     ecc_key  myKey;
                     ecc_key* peerKey = NULL;
-                    word32   size = MAX_ENCRYPT_SZ;
+                    word32   pubKeySz = MAX_ENCRYPT_SZ;
+                    word32   secSz = sizeof(ssl->arrays->preMasterSecret)
```

Comment:
Using the pre master secret size define is usually better.  Some of these array variables can be altered to pointers during optimization rounds and sizeof(pointer) becomes the wrong size.